JsonPatch/PatchExtensions.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Json.More;
+using Json.Pointer;
+
+namespace Json.Patch
+{
+	/// <summary>
+	/// Provides extension methods for <see cref="JsonPatch"/>.
+	/// </summary>
+	public static class PatchExtensions
+	{
+		/// <summary>
+		/// Apply a patch to an object to transform its data.
+		/// </summary>
+		/// <param name="patch">The patch</param>
+		/// <param name="obj">The object</param>
+		/// <typeparam name="T">The type of the object.</typeparam>
+		/// <returns>New instance of patched object</returns>
+		/// <exception cref="InvalidOperationException">Thrown when the patch cannot be applied.</exception>
+		public static T Apply<T>(this JsonPatch patch, T obj)
+		{
+			return Apply<T, T>(patch, obj);
+		}
+
+		/// <summary>
+		/// Apply a patch to an object to transform it to another type.
+		/// </summary>
+		/// <param name="patch">The patch</param>
+		/// <param name="obj">The object</param>
+		/// <param name="options">Serializer options.</param>
+		/// <typeparam name="TOriginal">The type of the original object.</typeparam>
+		/// <typeparam name="TTarget">The type of the target object.</typeparam>
+		/// <returns>New instance of patched object</returns>
+		/// <exception cref="InvalidOperationException">Thrown when the patch cannot be applied.</exception>
+		public static TTarget Apply<TOriginal, TTarget>(this JsonPatch patch, TOriginal obj, JsonSerializerOptions? options = null)
+		{
+			using var doc = JsonDocument.Parse(JsonSerializer.Serialize(obj, options));
+			var patchResult = patch.Apply(doc.RootElement);
+			if (!patchResult.IsSuccess)
+				throw new InvalidOperationException($"{patchResult.Error} Operation: {patchResult.Operation}");
+			var result = JsonSerializer.Deserialize<TTarget>(patchResult.Result.GetRawText());
```

Comment:
Deserialization need options too, just in case, I guess.

Reply:
Good catch.  I hope one options will suffice for both operations...

Reply:
I hope so too, but if it's optional may be splitting could be more convenient?

Reply:
I'm not fussed about it... most options are useful for one or the other direction, but not both.

Reply:
I'm just assuming that we could got Source from one library and Target to other library object, so we may need different converters... I guess it would be rare situation but if would be - single options could be not such goot thing...
I think you could use something like this
`var result = JsonSerializer.Deserialize<TTarget>(patchResult.Result.GetRawText(), deserializeOptions??options);`
Just suggesting, not infusing :)